### PR TITLE
fix: Fix geometric digitization of generic detector

### LINKED ISSUE #2020: bug: default geometric smearing for generic detector is broken
Trying to configure the digitization of the generic detector with the default config (default-geometric-config-generic.json) currently leads to the following error:

```
Traceback (most recent call last):
  File "/home/glg/ACTS/worktrees/workspaces/pos-calibrate/pos-calibrate.py", line 103, in <module>
    main()
  File "/home/glg/ACTS/worktrees/workspaces/pos-calibrate/pos-calibrate.py", line 100, in main
    generate_trackstates(nevents=5000, rng=rng,)
  File "/home/glg/ACTS/worktrees/workspaces/pos-calibrate/pos-calibrate.py", line 52, in generate_trackstates
    acts.examples.simulation.addDigitization(
  File "/home/glg/ACTS/worktrees/pos-calibrate/build/python/acts/examples/simulation.py", line 718, in addDigitization
    acts.examples.readDigiConfigFromJson(
RuntimeError: [json.exception.type_error.305] cannot use operator[] with a string argument with binary
```
I traced it back to #1834 . This PR introduced a configurable charge smearer, and reads it unconditionally from the config. However the default configuration was not remade, leading to this error.

One possible fix is the following patch:
```
modified   Examples/Io/Json/src/JsonDigitizationConfig.cpp
@@ -138,7 +138,9 @@ void ActsExamples::from_json(const nlohmann::json& j,
   gdc.thickness = j["thickness"];
   gdc.threshold = j["threshold"];
   gdc.digital = j["digital"];
-  from_json(j["charge-smearing"], gdc.chargeSmearer);
+  if (j.contains("charge-smearing")) {
+      from_json(j["charge-smearing"], gdc.chargeSmearer);
+  }
 }
 
 void ActsExamples::to_json(nlohmann::json& j,

```

Alternatively the default config file could simply be remade, but I don't know how to do that.

Pinging @benjaminhuth @asalzburger @paulgessinger , thoughts? If the above fix is fine, I can submit a PR but otherwise one of you needs to remake the default config (or teach me how to do it).

### PR DESCRIPTION
**Pull request by benjaminhuth**
Also adds this to CI tests. Should be also tested for ODD, but maybe in a separate PR.

Fixes #2020

### CODE REVIEW

**Comment by gagnonlg** [comment]
@benjaminhuth is there any showstopper for this MR? I'd like to stop having to include my hotfix from #2020 in every build :grin:

**Comment by gagnonlg** [comment]
Awesome, thanks @benjaminhuth

**Review by gagnonlg — APPROVED** [review]
Looks good to me!

File: Examples/Scripts/Python/exatrkx.py [hunk 1]
```diff
@@ -54,11 +61,12 @@
     rnd = acts.examples.RandomNumbers()
     outputDir = Path(os.getcwd())
 
-    s = configureDigitization(
+    s = runDigitization(
         trackingGeometry,
         field,
         outputDir,
         inputParticlePath,
+        digiConfigFile=digiConfigFile,
```
Comment (andiwand) [inline Examples/Scripts/Python/exatrkx.py:69]:
```suggestion
        digiConfigFile=digiConfigFile,
        particlesInput=inputParticlePath,
```

**Review by andiwand — APPROVED** [review]

**Pull request merged**